homeassistant/components/landisgyr_heat_meter/config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Config flow for Landis+Gyr Heat Meter integration."""
+from __future__ import annotations
+
+import logging
+import os
+
+import async_timeout
+import serial
+import serial.tools.list_ports
+from ultraheat_api import HeatMeterService, UltraheatReader
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ultraheat Heat Meter."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Step when setting up serial configuration."""
+        errors = {}
+
+        if user_input is not None:
+            user_selection = user_input[CONF_DEVICE]
+            if user_selection == CONF_MANUAL_PATH:
+                return await self.async_step_setup_serial_manual_path()
+
+            dev_path = await self.hass.async_add_executor_job(
+                get_serial_by_id, user_selection
+            )
+
+            try:
+                model, device_number = await self.validate_ultraheat(dev_path)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                await self.async_set_unique_id(device_number)
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=model,
+                    data=user_input | {"model": model, "device_number": device_number},
+                )
+
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        list_of_ports = {}
+        for port in ports:
+            list_of_ports[
+                port.device
+            ] = f"{port}, s/n: {port.serial_number or 'n/a'}" + (
+                f" - {port.manufacturer}" if port.manufacturer else ""
+            )
+        list_of_ports[CONF_MANUAL_PATH] = CONF_MANUAL_PATH
+
+        schema = vol.Schema({vol.Required(CONF_DEVICE): vol.In(list_of_ports)})
```

Comment:
Use a select selector instead, maybe something like:
```py
        port_options = [
            selector.SelectOptionDict(
                value=port.device,
                label=f"{port}, s/n: {port.serial_number or 'n/a'}"
                + (f" - {port.manufacturer}" if port.manufacturer else ""),
            )
            for port in ports
        ]
        port_options.append(
            selector.SelectOptionDict(value=CONF_MANUAL_PATH, label=CONF_MANUAL_PATH)
        )
        schema = vol.Schema(
            {
                vol.Required(CONF_MANUAL_PATH): selector.SelectSelector(
                    selector.SelectSelectorConfig(options=port_options),
                )
            }
        )
```

Reply:
I'll look into this. What is the benefit that's gained from this refactor?

Reply:
IIRC the difference is that the order of the selections is deterministic if the select selector.
It's fine to keep the "old fashioned" `vol.Schema({vol.Required(CONF_DEVICE): vol.In(list_of_ports)})`

Reply:
Ah, I understand. I looked at several other integrations, like rfxtrx, which have been out there for a while. I'll have a look at the selector, but good to know that it's not blocking.